File: src/Bionic.Bridge.Capacitor.JS/src/StatusBarBridge.ts
import {Plugins, StatusBarStyle} from "@capacitor/core";

const {StatusBar} = Plugins;

interface IStatusBar {
  show(): Promise<void>;
  hide(): Promise<void>;
  setBackgroundColor(color: string): Promise<void>;
  setStyle(style: number): Promise<void>
}

export class StatusBarBridge implements IStatusBar {
  async show(): Promise<void> {
    if (!StatusBar) return Promise.reject("StatusBar not available.");
    return await StatusBar.show();
  }

  async hide(): Promise<void> {
    if (!StatusBar) return Promise.reject("StatusBar not available.");
    return await StatusBar.hide();
  }

  async setBackgroundColor(color: string): Promise<void> {
    if (!StatusBar) return Promise.reject("StatusBar not available.");
    return await StatusBar.setBackgroundColor({color});
  }

  async setStyle(style: number): Promise<void> {
    if (!StatusBar) return Promise.reject("StatusBar not available.");
    return await StatusBar.setStyle({style: style === 0 ? StatusBarStyle.Dark : StatusBarStyle.Light});
  }
}

File: src/Bionic.Bridge.Capacitor/StatusBarBridge.cs
using System.Threading.Tasks;
using Microsoft.JSInterop;

namespace Bionic.Bridge.Capacitor {
    public static class StatusBarBridge {
        private const string StatusBarShowFunctionName = "BionicBridge.Capacitor.StatusBar.show";
        private const string StatusBarHideFunctionName = "BionicBridge.Capacitor.StatusBar.hide";

        private const string StatusBarSetBackgroundColorFunctionName =
            "BionicBridge.Capacitor.StatusBar.setBackgroundColor";

        private const string StatusBarSetStyleFunctionName = "BionicBridge.Capacitor.StatusBar.setStyle";

        public static Task Show() =>
            JSRuntime.Current.InvokeAsync<object>(StatusBarShowFunctionName);

        public static Task Hide() =>
            JSRuntime.Current.InvokeAsync<object>(StatusBarHideFunctionName);

        public static Task SetBackgroundColor(string color) =>
            JSRuntime.Current.InvokeAsync<object>(StatusBarSetBackgroundColorFunctionName, color);

        public static Task SetStyle(StatusBarStyle style) =>
            JSRuntime.Current.InvokeAsync<object>(StatusBarSetStyleFunctionName, style);
    }

    public enum StatusBarStyle {
        Dark,
        Light
    }
}